### PR TITLE
refactor(button-group): Fix the double borders between the buttons

### LINKED ISSUE #4091: Button Group REFACTORING
Fix the double borders between the buttons

### PR DESCRIPTION
**Pull request by desig9stein**
Closes #4091 

### Additional information (check all that apply):
 - [ ] Bug fix
 - [ ] New functionality
 - [ ] Documentation
 - [ ] Demos
 - [ ] CI/CD

### Checklist:
 - [x] All relevant tags have been applied to this PR
 - [ ] This PR includes unit tests covering all the new code
 - [ ] This PR includes API docs for newly added methods/properties
 - [ ] This PR includes `feature/README.MD` updates for the feature docs
 - [ ] This PR includes general feature table updates in the root `README.MD`
 - [ ] This PR includes `CHANGELOG.MD` updates for newly added functionality
 - [ ] This PR contains breaking changes
 - [ ] This PR includes `ng update` migrations for the breaking changes
 - [ ] This PR includes behavioral changes and the feature specification has been updated with them

### CODE REVIEW

File: projects/igniteui-angular/src/lib/core/styles/components/button-group/_button-group-theme.scss [hunk 1]
```diff
@@ -76,35 +79,23 @@
...
     }
 
     @if not($item-selected-text-color) and $item-selected-background {
-        $item-selected-text-color: text-contrast($item-background);
+        $item-selected-text-color: text-contrast($item-selected-background);
     }
 
-    @if not($disabled-text-color) and $item-background {
+    @if not($disabled-text-color) and $disabled-background-color {
         $disabled-text-color: rgba(text-contrast($item-background), .3);
```
Comment (simeonoff) [inline projects/igniteui-angular/src/lib/core/styles/components/button-group/_button-group-theme.scss]: Refactor to support gradients/images as values for the background.

File: projects/igniteui-angular/src/lib/core/styles/components/button-group/_button-group-theme.scss [hunk 2]
```diff
@@ -150,85 +139,97 @@
...
 
-        &:last-child {
-            border-right-width: 0;
+        &:last-of-type {
+            border-radius: 0 --var($theme, 'border-radius') --var($theme, 'border-radius') 0;
+        }
+
+        &%igx-button--disabled {
+            color: --var($theme, 'disabled-text-color') !important;
+            background-color: --var($theme, 'disabled-background-color') !important;
```
Comment (simeonoff) [inline projects/igniteui-angular/src/lib/core/styles/components/button-group/_button-group-theme.scss]: Use background in place of background-color.

File: projects/igniteui-angular/src/lib/core/styles/components/button-group/_button-group-theme.scss [hunk 3]
```diff
@@ -150,85 +139,97 @@
...
+            border-color: --var($theme, 'item-disabled-border');
         }
 
         &:hover,
-        &:focus{
+        &:focus {
             color: --var($theme, 'item-hover-text-color');
             background-color: --var($theme, 'item-hover-background');
```
Comment (simeonoff) [inline projects/igniteui-angular/src/lib/core/styles/components/button-group/_button-group-theme.scss]: Same as the one above.

File: projects/igniteui-angular/src/lib/core/styles/components/button-group/_button-group-theme.scss [hunk 4]
```diff
@@ -150,85 +139,97 @@
...
+        }
+
+        &:last-of-type {
+            border-radius: 0 0 --var($theme, 'border-radius') --var($theme, 'border-radius');
         }
     }
 
     %igx-group-item-selected {
         color: --var($theme, 'item-selected-text-color');
         background-color: --var($theme, 'item-selected-background');
```
Comment (simeonoff) [inline projects/igniteui-angular/src/lib/core/styles/components/button-group/_button-group-theme.scss]: Same as the one above.

File: projects/igniteui-angular/src/lib/core/styles/components/button-group/_button-group-theme.scss [hunk 5]
```diff
@@ -150,85 +139,97 @@
...
+        position: relative;
+        z-index: 1;
 
         &:hover,
-        &:focus{
+        &:focus {
             color: --var($theme, 'item-selected-text-color');
-            background-color: --var($theme, 'item-selected-background');
-            border-color: --var($theme, 'item-selected-border-color');
+            background-color: --var($theme, 'item-selected-hover-background');
```
Comment (simeonoff) [inline projects/igniteui-angular/src/lib/core/styles/components/button-group/_button-group-theme.scss]: And again.

File: projects/igniteui-angular/src/lib/core/styles/themes/schemas/light/_button-group.scss [hunk 6]
```diff
@@ -48,18 +57,21 @@ $_light-button-group: (
...
         igx-color: ('grays', 400)
     ),
 
     disabled-background-color: (
-        igx-color: ('grays', 100)
+        igx-color: ('grays', 50)
     ),
 
-    elevation: 2
+    elevation: 0
```
Comment (simeonoff) [inline projects/igniteui-angular/src/lib/core/styles/themes/schemas/light/_button-group.scss]: Doesn't this introduce too big of a visual breaking change? I'd keep as it was before.

**Comment by desig9stein** [comment]
@simeonoff before I do all the changes I will wait for your PR for the gradients backgrounds to be merged

**Pull request merged**